projects/storefrontlib/src/cms-components/configurator/commons/config-add-to-cart-button/config-add-to-cart-button.component.ts

```diff
@@ -106,7 +104,7 @@ export class ConfigAddToCartButtonComponent implements OnInit {
     const isOverview = pageType === ConfigurationRouter.PageType.OVERVIEW;
     const owner = configuration.owner;
     this.configRouterExtractorService
-      .extractRouterData(this.routingService)
+      .extractRouterData()
```

Comment:
I think `onAddToCart` might be doing too much. 
Do you want to discuss how we can refactor it?
And at the same time avoid the subscription, if possible.